### PR TITLE
PRL-6379: - Case party details 2

### PR DESCRIPTION
**Pull request by abigailsmith97**
[PRL-6379](https://tools.hmcts.net/jira/browse/PRL-6379)
List of pages: 

applicant/string/address/manual
applicant/string/contact-detail
applicant/string/contact-preference
respondent-details/add-respondents
User(s) to complete event: Citizen

Definition of done:

All POM sections are created
All POMs are added to a relevant journey

### CODE REVIEW

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantAddressSelectPage.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export class ApplicantAddressSelectPage {
       ),
       Helpers.checkVisibleAndPresent(
         page,
-        `${Selectors.GovukErrorMessage}:text-is("${ApplicantAddressSelectContent.errorMessage}")`,
+        `${Selectors.GovukErrorMessageCitizen}:text-is("${ApplicantAddressSelectContent.errorMessage}")`,
```
Comment (OLT2000) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantAddressSelectPage.ts:99]: I think this is a duplicate selector. There's one called ErrorMessage which is also `.govuk-error-message`

**Review by OLT2000 — DISMISSED** [review]
lgtm

**Review by Scralfie — APPROVED** [review]

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    if (accessibilityTest) {
+      await AccessibilityTestHelper.run(page);
+    }
+  }
+
+  private static async triggerErrorMessages(page: Page): Promise<void> {
+    await page.click(
+      `${Selectors.GovukButton}:text-is("${CommonStaticText.continue}")`,
+    );
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts:98]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+      ),
+      Helpers.checkGroup(
+        page,
+        3,
+        ApplicantContactDetailContent,
+        "errorMessage",
+        Selectors.GovukErrorMessageCitizen,
+      ),
+    ]);
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts:121]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+      ),
+      Helpers.checkGroup(
+        page,
+        2,
+        ApplicantContactDetailContent,
+        "hiddenErrorMessage",
+        Selectors.GovukErrorMessageCitizen,
+      ),
+    ]);
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts:145]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+      ),
+      Helpers.checkGroup(
+        page,
+        2,
+        ApplicantContactDetailContent,
+        "invalidErrorMessage",
+        Selectors.GovukErrorMessageCitizen,
+      ),
+    ]);
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactDetailPage.ts:171]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  private static async checkPageLoads({
+    page,
+    accessibilityTest,
+  }: checkPageLoadsOptions): Promise<void> {
+    await page.waitForSelector(
+      `${Selectors.GovukHeadingXL}:has-text("${ApplicantContactPreferenceContent.pageTitle}")`,
+    );
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts:61]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        `${uniqueSelectors.listSelector3}:text-is("${ApplicantContactPreferenceContent.li3}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${uniqueSelectors.legendSelector}:text-is("${ApplicantContactPreferenceContent.legend}")`,
+        1,
+      ),
+    ]);
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts:105]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    if (accessibilityTest) {
+      await AccessibilityTestHelper.run(page);
+    }
+  }
+
+  private static async triggerErrorMessages(page: Page): Promise<void> {
+    await page.click(
+      `${Selectors.GovukButton}:text-is("${CommonStaticText.continue}")`,
+    );
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/applicantContactPreferencePage.ts:115]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  private static async checkPageLoads({
+    page,
+    accessibilityTest,
+  }: CheckPageLoadsOptions): Promise<void> {
+    await page.waitForSelector(
+      `${Selectors.GovukHeadingXL}:text-is("${RespondentDetailsAddRespondentsContent.pageTitle}")`,
+    );
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts:54]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        `${UniqueSelectors.h2Selector}:text-is("${RespondentDetailsAddRespondentsContent.h2}")`,
+        1,
+      ),
+      Helpers.checkVisibleAndPresent(
+        page,
+        `${Selectors.GovukHint}:text-is("${RespondentDetailsAddRespondentsContent.hint}")`,
+        1,
+      ),
+    ]);
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts:79]: Whitespace

File: e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    if (accessibilityTest) {
+      await AccessibilityTestHelper.run(page);
+    }
+  }
+
+  private static async triggerErrorMessages(page: Page): Promise<void> {
+    await page.click(
+      `${Selectors.GovukButton}:text-is("${CommonStaticText.continue}")`,
+    );
+
```
Comment (JakeCG) [inline e2e/pages/citizen/createCase/C100/casePartyDetails/respondentDetailsAddRespondentsPage.ts:89]: Whitespace

**Review by JakeCG — CHANGES_REQUESTED** [review]

**Review by Scralfie — APPROVED** [review]

**Pull request merged**